### PR TITLE
Github5592

### LINKED ISSUE #5592: New bond highlighting code does a rubbish job with cyclopropane
**Describe the bug**
It's not unattractive, IMO, but clearly wrong.
image
**To Reproduce**
Steps to reproduce the behavior, feel free to paste a bit of Python in here.

**Expected behavior**
Plain rectangles under the cyclopropane bonds.

**Screenshots**

**Configuration (please complete the following information):**
 - RDKit version: Build from current master
 - OS: MacOS and Windows
 - Python version (if relevant):
 - Are you using conda?
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit?

**Additional context**
I'll start work on a fix.

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Fixes #5592

#### What does this implement/fix? Explain your changes.
Uses a convex hull algorithm to get the points for the bond highlight polygons in the right order for drawing.

#### Any other comments?
Tested on macOS only.

### CODE REVIEW

File: Code/GraphMol/MolDraw2D/DrawMol.cpp [hunk 1]
```diff
@@ -2053,16 +2053,37 @@ void DrawMol::makeBondHighlightLines(double lineWidth, double scale) {
...
+                      auto &p0 = points.front();
+                      auto val = (p1.y - p0.y) * (p2.x - p1.x) -
+                                 (p1.x - p0.x) * (p2.y - p1.y);
+                      if (val == 0.0) {
+                        return (p0 - p2).lengthSq() < (p0 - p1).lengthSq();
+                      } else if (val < 0.0) {
+                        return true;
+                      } else {
+                        return false;
+                      }
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawMol.cpp:2085]: A question: you're doing the comparisons here assuming exact math - i.e. `val == 0.0` or `p1.y < p2.y` - but you are working with doubles; would it be better to use approximate comparisons? 
I'm not sure if it's necessary, but figured it's worth thinking about.

**Review by greglandrum — COMMENTED** [review]
One question in the review.

File: Code/GraphMol/MolDraw2D/DrawMol.cpp (hunk 1 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/DrawMol.cpp:2085]: That is an interesting question.  Graham's scan normally has a second step that goes round turfing out all the inner points so the convex hull doesn't include them, and I think that that takes care of those sorts of issues.  In this case, we know we are going to have at most a hexagon with defined angles (long with pointy ends) so I don't think the problem will arise.  I can put the 2nd stage in if you like, or document its absence.

**Comment by DavidACosgrove** [comment]
The ubuntu failure is because of a checksum error with the inchi download.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
Thanks Dave.

**Pull request merged**